extensions/blocks/conversation/edit.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * External dependencies
+ */
+import { filter, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	useEffect,
+	useRef,
+	useCallback,
+	useMemo,
+} from '@wordpress/element';
+import {
+	InnerBlocks,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import {
+	Panel,
+	PanelBody,
+	ToggleControl,
+	ToolbarGroup,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import ParticipantsDropdown, { ParticipantsSelector } from './components/participants-controls';
+import TranscritptionContext from './components/context';
+
+import { list as defaultParticipants } from './participants.json';
+
+const TRANSCRIPTION_TEMPLATE = [
+	[ 'core/heading', { placeholder: __( 'Conversation title', 'jetpack' ) } ],
+	[ 'jetpack/dialogue', defaultParticipants[ 0 ] ],
+	[ 'jetpack/dialogue', defaultParticipants[ 1 ] ],
+	[ 'jetpack/dialogue', defaultParticipants[ 2 ] ],
+];
+
+function ConversationEdit ( {
+	className,
+	attributes,
+	setAttributes,
+} ) {
+	const { participants, showTimestamps, className: classNameAttr } = attributes;
+	const containertRef = useRef();
+
+	// Set initial conversation participants.
+	useEffect( () => {
+		if ( participants ) {
+			return;
+		}
+
+		setAttributes( { participants: defaultParticipants } );
+	}, [ participants, setAttributes ] );
+
+	const updateParticipants = useCallback( ( updatedParticipant ) => (
+		setAttributes( { participants: map( participants, ( participant ) => {
+			if ( participant.participantSlug !== updatedParticipant.participantSlug ) {
+				return participant;
+			}
+			return {
+				...participant,
+				...updatedParticipant,
+			};
+		} ) } )
+	), [ setAttributes, participants ] );
+
+	// Context bridge.
+	const contextProvision = {
+		setAttributes: useMemo( () => setAttributes, [ setAttributes ] ),
+		updateParticipants,
+
+		attributes: {
+			showTimestamps,
+			classNameAttr,
+		},
+	};
+
+	function deleteParticipant( deletedParticipantSlug ) {
+		setAttributes( { participants: filter( participants, ( { participantSlug } ) => ( participantSlug !== deletedParticipantSlug ) ) } );
+	}
+
+	function addNewParticipant( newSpakerValue ) {
+		const newParticipantSlug = participants.length
+			? ( participants[ participants.length - 1 ].participantSlug ).replace( /(\d+)/, ( n ) => Number( n ) + 1 )
+			: 'sepaker-0';
```

Comment:
Nitpick and non-blocking: There is another typo.
```suggestion
			: 'speaker-0';
```

Reply:
Also, should this be renamed to `participant-0` so it's aligned with the value of the default participants defined in the json file below?